src/pubkey.cpp

```diff
@@ -227,12 +227,13 @@ bool XOnlyPubKey::IsFullyValid() const
     return secp256k1_xonly_pubkey_parse(secp256k1_context_static, &pubkey, m_keydata.data());
 }
 
-bool XOnlyPubKey::VerifySchnorr(const uint256& msg, std::span<const unsigned char> sigbytes) const
+bool XOnlyPubKey::VerifySchnorr(
+    const std::span<const unsigned char> msg, std::span<const unsigned char> sigbytes) const
```

Comment:
needs to be asserting that msg.size() > 0

Reply:
secp256k1_schnorrsig_verify will assert otherwise

Reply:
nvm misread the ARGCHECK --> it's only checked that if msg is NULL, then length must be 0, which... makes sense!